### PR TITLE
Added GeoGebraTube Injection

### PR DESCRIPTION
**Pull request by daFischer**
Tested successfully with Firefox and Chromium.
Opera doesn't work with GGT yet.

Safari and Internet Explorer still have to be tested.

### CODE REVIEW

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 1]
```diff
@@ -9,14 +9,21 @@
...
         cache = {},
         ggbApplets = {},
         ggbAppletsCount = 0,
         geogebraScriptSource = 'https://www.geogebra.org/web/4.4/web/web.nocache.js',
-        $geogebraTemplate = $('<article class="geogebraweb" data-param-width="700" data-param-height="525" data-param-usebrowserforjs="true" data-param-enableRightClick="false"></article>');
+        $geogebraTemplate = $('<article class="geogebraweb" data-param-width="700" data-param-height="525" data-param-usebrowserforjs="true" data-param-enableRightClick="false"></article>'),
+        //gtApplets = {},
+        //geogebraTubeScriptSource = 'http://www.geogebratube.org/scripts/deployggb.js',
+        gtAppletsCount = 0,
+        $geogebraTubeTemplate = $('<div style="width:100% overflow:hidden"></div>');
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: missing comma in `"width:100% overflow:hidden"></div>');
Markdown supported `

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 2]
```diff
@@ -68,6 +75,40 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
                 }
             }
 
+            function initGeogebraTube() {
+                //console.log(href.substr(0, 34));
+                if (href.substr(0, 34) !== "http://tube.geogebra.org/student/m") {
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: Is it always `http://`? Maybe check against the hostname instead?

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 3]
```diff
@@ -68,6 +75,40 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
...
+                    return false;
+                }
+                var gtAppletID = 'gtApplet' + gtAppletsCount, applet, transform,
+                    $clone = $geogebraTubeTemplate.clone();
+
+                //if (gtAppletsCount === 0)
+                //    require([geogebraTubeScriptSource]);
+                gtAppletsCount += 1;
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: `gtAppletsCount++;`

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 4]
```diff
@@ -68,6 +75,40 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
...
+
+                $clone.attr('id', gtAppletID);
+
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: remove blank lines

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 5]
```diff
@@ -80,21 +121,23 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
...
+                if (! initGeogebraTube()) {
+                    if (data.documentElement && data.documentElement.nodeName === 'geogebra') {
+                        initGeogebraApplet(data.documentElement.outerHTML);
+                    } else if (contentType === 'image/jpeg' || contentType === 'image/png') {
+                        $that.html('<img src="' + href + '" title="' + title + '" />');
+                    } else {
+                        try {
+                            data = JSON.parse(data);
+                            if (data.response) {
+                                $that.html(data.response);
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: Why did you remove the markup: `$that.html('<div class="panel panel-default"><div class="panel-body">' + data.response + '</div></div>');` ?

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 6]
```diff
@@ -68,6 +75,40 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
...
+                applet.inject(gtAppletID, 'preferHTML5');
+
+                transform = function () {
+                    //$clone.find("div:first").css("transform", "none");
+                    $clone.parent().width($clone.find("div:first").width() * $clone.find("div:first > article").attr("data-param-scale"));
+                    $clone.parent().height($clone.find("div:first").height() * $clone.find("div:first > article").attr("data-param-scale"));
+                    //$clone.parent().css("overflow", "hidden");
+                    //console.log($clone.find("div:first > article").attr("data-param-scale"));
+                };
+                setTimeout(transform, 20000);
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: 20 Seconds timeout? What is that for?

File: src/assets/source/scripts/modules/serlo_injections.js [hunk 7]
```diff
@@ -80,21 +121,23 @@ define(['jquery', 'common', 'translator', 'content'], function ($, Common, t) {
...
-                } else if (contentType === 'image/jpeg' || contentType === 'image/png') {
-                    $that.html('<img src="' + href + '" title="' + title + '" />');
-                } else {
-                    try {
-                        data = JSON.parse(data);
-                        if (data.response) {
-                            $that.html('<div class="panel panel-default"><div class="panel-body">' + data.response + '</div></div>');
-                            Common.trigger('new context', $that);
-                        } else {
+                if (! initGeogebraTube()) {
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: What is this doing? Most injections don't need Geogebra?

File: src/assets/source/scripts/modules/serlo_injections.js (hunk 6 above)
Comment (daFischer) [inline src/assets/source/scripts/modules/serlo_injections.js]: sometimes it took pretty long for the GeoGebraTube Applet to load, so this is a precaution.
Maybe 20000 is longer than needed, but it shouldn't create any problems if the first timeout already worked.

Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: Isn't there something like a "ready" event you could listen on?

Comment (daFischer) [inline src/assets/source/scripts/modules/serlo_injections.js]: I've searched for it but found nothing.

File: src/assets/source/scripts/modules/serlo_injections.js (hunk 7 above)
Comment (daFischer) [inline src/assets/source/scripts/modules/serlo_injections.js]: I put the check for geogebratube into the function
It's bad style, so I'll refactor it.

File: src/assets/source/scripts/modules/serlo_injections.js (hunk 5 above)
Comment (daFischer) [inline src/assets/source/scripts/modules/serlo_injections.js]: Is it possible that the markup has been added recently?
The only possibility I could think of is that I worked with an older version.

Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_injections.js]: Yes that's possible.
